### ISSUE TITLE
Status Monitor somehow causes AmbiguousMatchException exception on AutoRest apps

### ISSUE THREAD

**Issue by SergeyKanzhelev** [Microsoft/ApplicationInsights-dotnet-server#241]
See https://github.com/Azure/autorest/issues/1542 for details.

**Comment by SergeyKanzhelev** [comment]
Ok. I took [this](https://github.com/SergeyKanzhelev/RTIA) app and inserted the code into the beginning:

``` csharp
Console.WriteLine("Number of custom attributes:" + System.Attribute.GetCustomAttributes((typeof(object)).Assembly).Length);

foreach(var a in System.Attribute.GetCustomAttributes((typeof(object)).Assembly))
{
    Console.WriteLine("attribute: " + a.TypeId);
}
```
Some attributes were duplicated and one was new:

```
attribute: System.CLSCompliantAttribute
attribute: System.Diagnostics.DebuggableAttribute
!dup: attribute: System.Diagnostics.DebuggableAttribute
attribute: System.Reflection.AssemblyCompanyAttribute
attribute: System.Reflection.AssemblyCopyrightAttribute
attribute: System.Reflection.AssemblyDefaultAliasAttribute
attribute: System.Reflection.AssemblyDelaySignAttribute
attribute: System.Reflection.AssemblyDescriptionAttribute
attribute: System.Reflection.AssemblyFileVersionAttribute
!dup: attribute: System.Reflection.AssemblyFileVersionAttribute
attribute: System.Reflection.AssemblyInformationalVersionAttribute
attribute: System.Reflection.AssemblyKeyFileAttribute
attribute: System.Reflection.AssemblyProductAttribute
attribute: System.Reflection.AssemblySignatureKeyAttribute
attribute: System.Reflection.AssemblyTitleAttribute
attribute: System.Resources.NeutralResourcesLanguageAttribute
attribute: System.Resources.SatelliteContractVersionAttribute
attribute: System.Runtime.CompilerServices.CompilationRelaxationsAttribute
!dup: attribute: System.Runtime.CompilerServices.CompilationRelaxationsAttribute
attribute: System.Runtime.CompilerServices.DefaultDependencyAttribute
attribute: System.Runtime.CompilerServices.ExtensionAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.InternalsVisibleToAttribute
attribute: System.Runtime.CompilerServices.RuntimeCompatibilityAttribute
!dup: attribute: System.Runtime.CompilerServices.RuntimeCompatibilityAttribute
attribute: System.Runtime.CompilerServices.StringFreezingAttribute
attribute: System.Runtime.InteropServices.ComCompatibleVersionAttribute
attribute: System.Runtime.InteropServices.ComVisibleAttribute
attribute: System.Runtime.InteropServices.GuidAttribute
attribute: System.Runtime.InteropServices.TypeLibVersionAttribute
!new: attribute: System.Runtime.Versioning.TargetFrameworkAttribute
attribute: System.Security.AllowPartiallyTrustedCallersAttribute
attribute: System.Security.Permissions.SecurityPermissionAttribute
attribute: System.Security.SecurityRulesAttribute
```

**Comment by SergeyKanzhelev** [comment]
Ok, it seems that those dups are coming from the assembly 'Microsoft.ApplicationInsights.Agent_x64.2.0.5\content\RTIA\x64\Microsoft.Diagnostics.Instrumentation.Extensions.Base.dll'. Here is the metadata of it:

```
[assembly: AssemblyVersion("2.0.5.0")]
[assembly: Debuggable(DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints)]
[assembly: AssemblyFileVersion("2.0.5.32465")]
[assembly: CompilationRelaxations(8)]
[assembly: RuntimeCompatibility(WrapNonExceptionThrows = true)]
[assembly: TargetFramework(".NETFramework,Version=v4.0,Profile=Client", FrameworkDisplayName = ".NET Framework 4 Client Profile")]
```

We need to remove attributes from this library when compiling it.

**Comment by SergeyKanzhelev** [comment]
I confirmed that removing of `TargetFramework` from the assembly will make it not to insert to `mscrolib`. I've used this property in csproj: `<GenerateTargetFrameworkAttribute>false</GenerateTargetFrameworkAttribute>`

So we just need to remove all other attributes and release a fix for Status Monitor and Azure Web App extension.

**Comment by dnduffy** [comment]
Does that mean we need to build a 2.0.7 version tonight/tomorrow?

**Comment by SergeyKanzhelev** [comment]
This issues is in StatusMonitor/MMA & Azure Web Site extension. Not in Intercept.

Please note, `GenerateTargetFrameworkAttribute` only removes one attribute. Work needs to be done to remove all other attributes.

**Comment by SergeyKanzhelev** [comment]
This should be fixed on Azure Web Apps extension [v2.2](https://www.siteextensions.net/packages/Microsoft.ApplicationInsights.AzureWebSites/2.2.0).

Update for the Status Monitor is pending

**Comment by SergeyKanzhelev** [comment]
This is completed
